src/transformers/models/detr/configuration_detr.py

```diff
@@ -169,6 +179,17 @@ def __init__(
         eos_coefficient=0.1,
         **kwargs
     ):
+        if not use_timm_backbone:
+            if backbone_config is None:
+                logger.info("`backbone_config` is `None`. Initializing the config with the default `ResNet` backbone.")
+                backbone_config = ResNetConfig(out_features=["stage2", "stage3", "stage4"])
+            elif isinstance(backbone_config, dict):
+                backbone_model_type = backbone_config.get("model_type")
+                config_class = CONFIG_MAPPING[backbone_model_type]
+                backbone_config = config_class.from_dict(backbone_config)
+
```

Comment:
There should be some safeguards here if the user passes both a `backbone_config` and `use_timm_backbone=True` as the `backbone_config` will then be completely ignored (so we should error to not surprise the user later).